docs/csharp/pattern-matching.md

```diff
@@ -1,268 +1,72 @@
+
+:::code language="csharp" source="snippets/patterns/Program.cs" ID="TypeCheckDisposable":::
+
+The same tests can be applied in a `switch` expression to test a variable against multiple different types. You can use that information to create better algorithms based on the specific run-time type.
+
+## Compare discrete values
+
+You can also test a variable to find a match on specific values. The following code shows one example where you test a value against all possible values declared in an enumeration:
+
+:::code language="csharp" source="snippets/patterns/Simulation.cs" ID="PerformOperation":::
+
+The previous example demonstrates a method dispatch based on the value of an enumeration. The final `_` case is a [*discard pattern*](language-reference/operators/patterns.md#discard-pattern) that matches all values. It handles any error conditions where the value doesn't match one of the defined `enum` values. If you omit that switch arm, the compiler warns that you haven't handled all possible input values. At run time, the `switch` expression throws an exception if the object being examined doesn't match any of the switch arms. You could use numeric constants instead of a set of enum values. You can also use this similar technique for constant string values that represent the commands:
+
+:::code language="csharp" source="snippets/patterns/Simulation.cs" ID="PerformStringOperation":::
+
+The preceding example shows the same algorithm, but uses string values instead of an enum. You would use this scenario if your application responds to text commands instead of a regular data format. In all these examples, the *discard pattern* ensures that you handle every input. The compiler helps you by making sure every possible input value is handled.
+
+## Relational patterns
+
+You can use [*relational patterns*](language-reference/operators/patterns.md#relational-patterns) to test how a value compares to constants. For example, the following code returns the state of water based on the temperature in Fahrenheit:
+
+:::code language="csharp" source="snippets/patterns/Simulation.cs" ID="RelationalPattern":::
+
+The preceding code also demonstrates the conjunctive `and` [*logical pattern*](language-reference/operators/patterns.md#logical-patterns) to check that both relational patterns match. You can also use a disjunctive `or` pattern to check that either pattern matches. The two relational patterns are surrounded by parentheses, which you can use around any pattern for clarity. The final two switch arms handle the cases for the melting point and the boiling point. Without those two arms, the compiler warns you that your logic doesn't cover every possible input.
+
+## Multiple inputs
+
+All the patterns you've seen so far have been checking one input. You can write patterns that examine multiple properties of an object. The following code examines the number of items and the value of an order to calculate a discounted price:
```

Comment:
This is the only example where there is an object that is being evaluated, and it made me ask myself before seeing it used - what the shape of the object is. It would be really helpful to first show the `Order` object and then show the pattern matching on its members as it is easier to follow from a mental model stand point.

> Show me the thing, then show me what you do with it.